SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPCQueues.h

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/RPCs/RPCService.h"
+
+namespace SpatialGDK
+{
+template <typename Payload, typename AdditionalSendingData = EmptyData>
+struct TRPCLocalOverflowQueue : public TRPCQueue<Payload, AdditionalSendingData>
+{
+	TRPCLocalOverflowQueue(FName InName, TRPCBufferSender<Payload>& Sender)
+		: TRPCQueue<Payload, AdditionalSendingData>(InName, Sender)
+	{
+	}
+
+	void FlushAll(RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&) override;
+	void Flush(Worker_EntityId EntityId, RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&,
+			   bool bIgnoreAdded = false) override;
+	void OnAuthLost(Worker_EntityId EntityId) override;
+	void OnAuthGained_ReadComponent(const RPCReadingContext& iCtx) override;
+
+protected:
+	bool FlushQueue(Worker_EntityId EntityId, typename TRPCQueue<Payload, AdditionalSendingData>::QueueData& Queue, RPCWritingContext& Ctx,
+					const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&);
+};
+
+template <typename Payload, typename AdditionalSendingData>
+struct TRPCFixedCapacityQueue : public TRPCLocalOverflowQueue<Payload, AdditionalSendingData>
+{
+	TRPCFixedCapacityQueue(FName InName, TRPCBufferSender<Payload>& Sender, uint32 InCapacity)
+		: TRPCLocalOverflowQueue<Payload, AdditionalSendingData>(InName, Sender)
+		, Capacity(InCapacity)
+	{
+	}
+
+	int32 Capacity;
+
+	void Push(Worker_EntityId EntityId, Payload&& Data, AdditionalSendingData&& AddData) override;
+	void FlushAll(RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&) override;
+	void Flush(Worker_EntityId EntityId, RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&,
+			   bool bIgnoreAdded = false) override;
+};
+
+template <typename Payload, typename AdditionalSendingData = EmptyData>
+struct TRPCMostRecentQueue : public TRPCFixedCapacityQueue<Payload, AdditionalSendingData>
+{
+	TRPCMostRecentQueue(FName InName, TRPCBufferSender<Payload>& Sender, uint32 InCapacity)
+		: TRPCFixedCapacityQueue<Payload, AdditionalSendingData>(InName, Sender, InCapacity)
+	{
+	}
+
+	void Push(Worker_EntityId EntityId, Payload&& Data, AdditionalSendingData&& AddData) override;
+};
+
+template <typename Payload, typename AdditionalSendingData>
+void TRPCLocalOverflowQueue<Payload, AdditionalSendingData>::FlushAll(
+	RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback& SentCallback)
+{
+	for (auto Iterator = this->Queues.CreateIterator(); Iterator; ++Iterator)
```

Comment:
Is there a reason for the explicit use of this here and elsewhere in these templates?

Reply:
Yeah, clang does not like referencing members that are in the parent class without it.